static/js/jira-connect-ghe.js

```diff
@@ -0,0 +1,63 @@
+const ALLOWED_PROTOCOLS = ["http:", "https:"];
+const GITHUB_CLOUD = ["github.com", "www.github.com"];
+
+/**
+ * Method that checks the validity of the passed URL
+ *
+ * @returns {boolean}
+ */
+const checkValidUrl = inputURL => {
+	try {
```

Comment:
Don't think we need to wrap this in a try/catch block. We really just want to return a boolean: if valid return true, if invalid return false. An invalid result will tell the UI to render the error.

Reply:
But why do we need to throw an error here? This should be a simple check to determine if the url is valid or not.

Reply:
To separate out the host and protocols and thus checking if passed string is a correct URL or not, I am using `new URL()`. This command throws a **JS error** in console for any invalid strings. So those invalid errors are being captured by `catch` which returns a false value.